feat: Add right-click context menu with Rename for Asset Browser

WangSimiao2000 opened the pull request:
Adds right-click context menu and Rename functionality to the Asset Browser, completing part of the TODO list.

1. Added right-click menu detection and display in `asset_browser_window.cpp`
2. Added member variables to `asset_browser_window.hpp`
3. Implemented rename modal dialog with:
   - Enter to confirm / Escape to cancel
   - Auto-focus on input field
   - Renames both asset file and corresponding

zzxzzk115 committed:
chore: fix README bad underline.

zzxzzk115 committed:
feat: clean code. handle scene view inputs: QWER controls.

zzxzzk115 committed:
fix: null entity if it's been deleted.

WangSimiao2000 committed:
Adds context menu to asset browser window

Introduces right-click context menu with options for reimport, create, delete, and rename actions to improve asset management workflow.
Prepares for future implementation of asset operations.

Signed-off-by: wangsimiao1 <wangsimiao1@xiaomi.com>

WangSimiao2000 committed:
Adds asset renaming functionality to browser

Enables users to rename assets via a modal popup in the asset browser UI.
Automatically handles renaming associated metadata files to maintain consistency.
Improves workflow by allowing direct asset management within the editor.

Signed-off-by: wangsimiao1 <wangsimiao1@xiaomi.com>

zzxzzk115 commented on source/editor/src/ui/windows/asset_browser_window.cpp:
```diff
@@ -80,6 +80,76 @@ namespace vultra
+                    if (ImGui::IsKeyPressed(ImGuiKey_Enter))
+                    {
+                        auto newPath = m_RenamingPath.parent_path() / m_RenameBuffer;
+                        std::filesystem::rename(m_RenamingPath, newPath);
```
Renaming assets directly won't update the asset registry. And here `m_RenameBuffer` doesn't care about the file extension, which is dangerous. The best way to implement Asset Renaming is to implement  AssetRegistry::rename first, then use this function to rename assets correctly. I'll update the AssetRegistry, then you can modify this pull request.

zzxzzk115 reviewed (changes requested):
Wait for AssetRegistry::rename.

zzxzzk115 committed:
feat: add implot.

zzxzzk115 committed:
bump libvultra version.

WangSimiao2000 committed:
Adds asset renaming functionality to browser

Enables users to rename assets via a modal popup in the asset browser UI.
Automatically handles renaming associated metadata files to maintain consistency.
Improves workflow by allowing direct asset management within the editor.

Signed-off-by: wangsimiao1 <wangsimiao1@xiaomi.com>

WangSimiao2000 committed:
Adds context menu to asset browser window

Introduces right-click context menu with options for reimport, create, delete, and rename actions to improve asset management workflow.
Prepares for future implementation of asset operations.

Signed-off-by: wangsimiao1 <wangsimiao1@xiaomi.com>

zzxzzk115 committed:
feat: use latest VAssetRegistry::updateRegistry. add AssetDatabase::renameAsset function. clean code.

zzxzzk115 committed:
Merge branch 'assets_browser' of github.com:WangSimiao2000/Vultra into WangSimiao2000-assets_browser

zzxzzk115 committed:
bump libvultra version to fix compiling error.

zzxzzk115 commented:
Added new functions: 
- AssetDatabase::renameAsset 
- VAssetRegistry::updateRegistry
- VAssetRegistry::unregisterAsset
